crates/forge/src/lib.rs

```diff
@@ -173,7 +179,16 @@ pub fn main_execution() -> Result<ExitStatus> {
 
             Ok(ExitStatus::Success)
         }
-        ForgeSubcommand::Test { args } => {
+        ForgeSubcommand::Test { mut args } => {
+            // clap can capture the first flag in test filter,
+            // let's move it to coverage_or_profiler_args
+            if let Some(test_filter) = &args.test_filter {
+                if test_filter.starts_with('-') {
+                    args.coverage_or_profiler_args.insert(0, test_filter.into());
+                    args.test_filter = None;
+                }
+            }
```

Comment:
I suppose that such heuristic can generate problems in the future, and introducing such condition disturbs the readability of the code. Maybe we should just add coverage subcommand to catch arguments in simpler and cleaner way? What do you think?

Reply:
> I think we could at least check if --coverage or --build-profile were actually passed. Because I think this heuristic should be only enabled in this case, otherwise, we shouldn't be doing anything with the filter.

yes, it's the correct one, but it is going to be the same thing as if there was just a subcommand.

Reply:
> I suppose that such heuristic can generate problems in the future, and introducing such condition disturbs the readability of the code. Maybe we should just add coverage subcommand to catch arguments in simpler and cleaner way? What do you think?

I guess we were discussing this earlier i don't know why we bring it here now and block it with the request changes 😅